File: DataDriven/dotNet/Entities/Comment.cs
using System.ComponentModel.DataAnnotations;

namespace dotNet.Entities;

class Comment
{
    [Key]
    public int Id { get; set; }

    [Required]
    public string AuthorName { get; set; }

    [Required]
    public string Content { get; set; }

    [Required]
    public DateTime LastUpdated { get; set; }

    [Required]
    public Post Post { get; set; }

}

File: DataDriven/Phoenix/app/lib/app/blog/comment.ex
defmodule App.Blog.Comment do
  use Ecto.Schema
  import Ecto.Changeset

  schema "comments" do
    field :authorName, :string
    field :content, :string
    belongs_to :post, App.Blog.Post
    field :lastUpdated, :naive_datetime
  end

  @doc false
  def changeset(comment, attrs) do
    comment
    |> cast(attrs, [:authorName, :content, :post, :lastUpdated])
    |> validate_required([:authorName, :content, :post, :lastUpdated])
  end
end
